### PR TITLE
Code-Coverage

### PR DESCRIPTION
**Pull request by phillipc**
The test run for our code coverage is complete! Good old karma-Istanbul/NYC skills are needed to get it running. 💪💪

## Summary by CodeRabbit

* **Chores**
  * Added flag-gated coverage support, a test-coverage target, and cleanup/ignore rules to exclude coverage artifacts and temp data.
* **Documentation**
  * Added a static coverage report and updated the README with test-coverage usage and related notes.
* **Tests**
  * Added and updated unit tests for lifecycle anchoring, binding preprocessing, descendants-complete behavior, and text interpolation.
* **Refactor**
  * Updated event-subscription API/behavior and adjusted subscription handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: COVERAGE.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+ packages/utils/spec                   |   94.65 |    55.31 |   94.51 |   94.94 |                   
+  arrayEditDetectionBehaviors.ts       |     100 |      100 |      90 |     100 |                   
+  domNodeDisposalBehaviors.ts          |   91.72 |      100 |   91.66 |   91.72 | ...09-214,224-231 
+  memoizationBehaviors.ts              |     100 |      100 |    92.3 |     100 |                   
+  onErrorBehaviors.ts                  |    87.8 |    33.33 |      90 |    87.8 | 19,24,37-43       
+  parseHtmlFragmentBehavior.ts         |    88.4 |    16.66 |     100 |    90.9 | 67-71,147,162     
+  taskBehaviors.ts                     |   98.67 |        0 |   96.66 |   98.66 | 144,303-304       
+  utilsBehaviors.ts                    |   99.28 |      100 |    95.4 |   99.28 | 167,309           
+  utilsDomBehaviors.ts                 |   78.26 |     64.7 |    90.9 |   79.12 | 58-90,101         
+
```
[the remark this answers is not in the document]
Comment (phillipc) [reply COVERAGE.md:177]: I want to put it in the repository for documentation and revision purposes.

File: tools/karma.conf.js [hunk 2]
```diff
@@ -2,14 +2,39 @@
...
+      const js = result.outputFiles.find(f => f.path.match(/\.js$/))
+      const sourceMap = result.outputFiles.find(f => f.path.match(/\.js\.map$/))
+      const sourceMapObject = JSON.parse(sourceMap.text)
+      sourceMapObject.sourceRoot = '/'
+
+      const instrumented = coverageInstrumenter.instrumentSync(js.text, null, sourceMapObject)
+      js.contents = Buffer.from(instrumented)
+    })
+  }
+}
```
[the remark this answers is not in the document]
Comment (phillipc) [reply tools/karma.conf.js:32]: Fixed!

File: packages/lifecycle/spec/LifeCycleBehaviors.ts [hunk 3]
```diff
@@ -211,5 +209,45 @@ describe('KO LifeCycle', function () {
...
+          this.addEventListener(div, 'click', () => divClick++)
+          this.addEventListener(anchor, 'click', () => anchorClick++)
+        }
+      }
+
+      const nlc = new NodeLifeCycle()
+      const nlc2 = new NodeLifeCycle()
+      assert.equal(o.getSubscriptionsCount(), 2)
+
+      nlc.anchorTo(nlc2)
```
Comment (phillipc) [inline packages/lifecycle/spec/LifeCycleBehaviors.ts:233]: @brianmhunt I have found this missing edge case in the coverage report and create a test case. Is it possible, that we don't need the nesting of lifecycle instances at all?
